airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/DestinationInitialState.java

```diff
@@ -0,0 +1,22 @@
+package io.airbyte.integrations.base.destination.typing_deduping;
+
+import java.util.Optional;
+
+/**
+ * Interface representing the initial state of a destination table.
+ *
+ * @param <TableDefinition> type of the table definition.
+ */
+public interface DestinationInitialState<TableDefinition> {
```

Comment:
2 nits:
* kotlin? :P
* why the interface+record? why not just have the record directly?

Reply:
Used it for an easy path to make unit test mock's `when..thenReturn` work with Intf.

Reply:
I feel like `mock(DestinationInitialState.class)` instead of `new DestinationInitialStateImpl(null, null, null, false, false)` isn't a _huge_ win? since it behaves exactly the same but is less explicit

unless you're saying `thenReturn` only accepts mockable objects

Reply:
yeah that works too. Just that i have more control in individual tests with `when...thenReturn` returning different outcomes on individual getters of that record rather than reconstructing the whole object with the desired outputs and rest as null.

Reply:
I'm leaving this as-is since you'll be changing this in your PR.